lib/matplotlib/tests/test_offsetbox.py

```diff
@@ -79,25 +79,39 @@ def test_offsetbox_clip_children():
     assert fig.stale
 
 
-def test_offsetbox_loc_codes():
-    # Check that valid string location codes all work with an AnchoredOffsetbox
-    codes = {'upper right': 1,
-             'upper left': 2,
-             'lower left': 3,
-             'lower right': 4,
-             'right': 5,
-             'center left': 6,
-             'center right': 7,
-             'lower center': 8,
-             'upper center': 9,
-             'center': 10,
-             }
-    fig, ax = plt.subplots()
-    da = DrawingArea(100, 100)
-    for code in codes:
-        anchored_box = AnchoredOffsetbox(loc=code, child=da)
+_loc_values = [['NW', 'N', 'NE', 'W', 'C', 'E', 'SW', 'S', 'SE'],
+               ['northwest', 'north', 'northeast', 'west', 'center',
+                'east', 'southwest', 'south', 'southeast'],
+               ['upper left', 'upper center', 'upper right', 'center left',
+                'center', 'right', 'lower left', 'lower center',
+                'lower right']]
+
+
+@pytest.mark.parametrize('locs', _loc_values)
+def test_offsetbox_loc_codes(locs):
```

Comment:
Should this become `@check_figures_equal`? Otherwise the drawing of the figures at the end is meaningless.

Reply:
This was intended as smoke test. A legend with an invalid `_loc` might be constructed, but would fail when being drawn. I don't think image comparisson would help here; however if you are concerned about drawing 6 figures taking too much time, I could externalize this part and run it only the necessary 4 times.

Reply:
No, 6 times is fine IMHO. Can you please add a short comment on the intent as you've described above?